be/src/storage/lake/replication_txn_manager.cpp

```diff
@@ -416,7 +423,14 @@ Status ReplicationTxnManager::convert_rowset_meta(const RowsetMeta& rowset_meta,
         std::string new_segment_filename = gen_segment_filename(transaction_id);
 
         rowset_metadata->add_segments(new_segment_filename);
-        auto pair = filename_map->emplace(std::move(old_segment_filename), std::move(new_segment_filename));
+        FileEncryptionInfo encryption_info;
+        if (config::enable_transparent_data_encryption) {
```

Comment:
is this var mutable? what if the configvar differs between different BE/CN, can this var move to FE as a cluster-level config item?

Thought it will be high risk of turning on/off this configvar, easily cause data corruption.

Reply:
no it's not, if enable_transparent_data_encryption is differ than fe conf, BE won't start. This var is added to enforce consistency with FE.  Before connecting to FE, BE may still do some task without FE, so a BE side config is neccesary.